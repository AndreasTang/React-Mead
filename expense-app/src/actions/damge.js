import uuid from 'uuid'

const addDamge = ({ description = '', note = '', amount = '', createdAt = 0 } = {}) => ({
    type: 'ADD_DAMGE',
    damge: {
        id: uuid(),
        description,
        note,
        amount,
        createdAt
    }
})

const removeDamge = ({ id } = {}) => ({
    type: 'REMOVE_DAMGE',
    id
})

const editDamge = (id, editDamge ) => ({
    type: 'EDIT_DAMGEAMOUNT',
    id,
    editDamge
})


export { addDamge, removeDamge, editDamge }