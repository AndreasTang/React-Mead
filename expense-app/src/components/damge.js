import React from 'react'
import DamgeList from './damgeList'
import DamgeListFilter from './damgeListFilter'

const Damge = () => (
    <div>
        <DamgeListFilter/>
        <DamgeList/>
    </div>
)

export default Damge