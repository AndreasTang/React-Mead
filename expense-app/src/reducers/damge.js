const damgeReducerDefaultState = []

const damgeReducer = (state = damgeReducerDefaultState, action) => {
    switch (action.type) {
        case 'ADD_DAMGE': {
            return [...state, action.damge]
        }
        case 'REMOVE_DAMGE': {
            return state.filter(({ id }) => id !== action.id)
        }
        case 'EDIT_DAMGEAMOUNT' : {
            return state.map((damge) => {
                if (damge.id === action.id) {
                    return {
                        ...damge,
                        ...action.editDamge
                    }
                } else {
                    return damge
                }
            })
        }
        default: 
            return state
    }
}

export default damgeReducer