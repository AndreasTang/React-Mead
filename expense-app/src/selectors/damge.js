import moment from 'moment'

const getVisibleDamge = (damge, { text, sortBy, startDate, endDate }) => {
    return damge.filter((damge) => {
        const startDateMatch = startDate ? startDate.isSameOrBefore(moment(damge.createdAt), 'day') : true
        const endDateMatch = endDate ? endDate.isSameOrAfter(moment(damge.createdAt), 'day') : true
        const textMatch = damge.description.toLowerCase().includes(text.toLowerCase())

        return startDateMatch && endDateMatch && textMatch
    }).sort((a, b) => {
        if (sortBy === 'date') {
            return a.createdAt > b.createdAt ? -1 : 1
        } else if (sortBy === 'amount') {
            return a.amount > b.amount ? -1 : 1
        }
    })
}

export default getVisibleDamge