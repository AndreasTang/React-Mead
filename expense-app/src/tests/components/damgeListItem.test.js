import React from 'react'
import { shallow } from 'enzyme'
import DamgeListItem from '../../components/damgeListItem'
import damge from '../fixtures/damge'

test('Testing should DamgeListItem render currectly' , () => {
    const wrapper = shallow(<DamgeListItem {...damge[0]} />)
    expect(wrapper).toMatchSnapshot()
})