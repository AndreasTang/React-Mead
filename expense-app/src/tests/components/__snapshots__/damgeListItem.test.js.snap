// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing should DamgeListItem render currectly 1`] = `
<div>
  <NavLink
    to="/edit/1"
  >
    <p>
      Description: 
      kicks
    </p>
  </NavLink>
  <p>
    Amount: 
    10
  </p>
  <p>
    CreateAt: 
    -432000000
  </p>
</div>
`;
